ci-operator/templates/openshift/installer/cluster-launch-installer-e2e-temp-new-test-binary.yaml

```diff
@@ -0,0 +1,940 @@
+          export KUBE_SSH_USER=core
+        elif [[ "${CLUSTER_TYPE}" == "azure4" ]]; then
+          export TEST_PROVIDER='azure'
+        fi
+
+        mkdir -p /tmp/output
+        cd /tmp/output
+
+        function setup_ssh_bastion() {
+          export SSH_BASTION_NAMESPACE=test-ssh-bastion
+          echo "Setting up ssh bastion"
+          mkdir -p ~/.ssh
+          cp "${KUBE_SSH_KEY_PATH}" ~/.ssh/id_rsa
+          chmod 0600 ~/.ssh/id_rsa
+          if ! whoami &> /dev/null; then
+            if [[ -w /etc/passwd ]]; then
+              echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+            fi
+          fi
+          curl https://raw.githubusercontent.com/eparis/ssh-bastion/master/deploy/deploy.sh | bash
+          for i in $(seq 0 30); do
+            # AWS fills only .hostname of a service
+            BASTION_HOST=$(oc get service -n "${SSH_BASTION_NAMESPACE}" ssh-bastion -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+            if [[ -n "${BASTION_HOST}" ]]; then break; fi
+            # Azure fills only .ip of a service. Use it as bastion host.
+            BASTION_HOST=$(oc get service -n "${SSH_BASTION_NAMESPACE}" ssh-bastion -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+            if [[ -n "${BASTION_HOST}" ]]; then break; fi
+            echo "Waiting for SSH bastion load balancer service"
+            sleep 10
+          done
+          if [[ -z "${BASTION_HOST}" ]]; then
+            echo "Failed to find bastion address, exiting"
+            exit 1
+          fi
+          export KUBE_SSH_BASTION="${BASTION_HOST}:22"
+        }
+
+        function setup-google-cloud-sdk() {
+          pushd /tmp
+          curl -O https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-256.0.0-linux-x86_64.tar.gz
+          tar -xzf google-cloud-sdk-256.0.0-linux-x86_64.tar.gz
+          export PATH=$PATH:/tmp/google-cloud-sdk/bin
+          mkdir gcloudconfig
+          export CLOUDSDK_CONFIG=/tmp/gcloudconfig
+          gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+          gcloud config set project openshift-gce-devel-ci
+          popd
+        }
+
+        function run-upgrade-tests() {
+          openshift-tests-kubernetes run-upgrade "${TEST_SUITE}" --to-image "${RELEASE_IMAGE_LATEST}" \
+            --options "${TEST_OPTIONS:-}" \
+            --provider "${TEST_PROVIDER:-}" -o /tmp/artifacts/e2e.log --junit-dir /tmp/artifacts/junit
+        }
+
+        function run-tests-kubernetes() {
+          openshift-tests-kubernetes run "${TEST_SUITE}" \
+            --provider "${TEST_PROVIDER:-}" -o /tmp/artifacts/e2e.log --junit-dir /tmp/artifacts/junit
+        }
+        function run-tests-extended() {
+          extended-platform-tests run "${TEST_SUITE}" \
+            --provider "${TEST_PROVIDER:-}" -o /tmp/artifacts/e2e.log --junit-dir /tmp/artifacts/junit
+        }
```

Comment:
atm, the above 2 fns run-tests-kubernetes and run-tests-extended run identical tests, and the run-upgrade-tests fn runs openshift-tests-kubernetes binary. (TEST_SUITE is set in the job yaml)